docs/developer/solutions/introduction.rst

```diff
@@ -0,0 +1,115 @@
+- Optionally, a UI for managing and monitoring the application, represented by
+  a standard Kubernetes ``Deployment``
+
+.. _Operator: https://coreos.com/blog/introducing-operators.html
+
+For more details, see the following documentation pages:
+
+- :doc:`./archive`
+- :doc:`./operator`
+- (TODO) Solution UI guidelines
+
+Once a Solution is deployed on MetalK8s, a user can deploy one or more versions
+of the Solution Operator, using either the Solution UI or the Kubernetes API,
+into separate namespaces. Using the Operator-defined ``CustomResource(s)``, the
+user can then effectively deploy the application packaged in the Solution.
+
+How is a *Solution* declared in MetalK8s?
+-----------------------------------------
+
+MetalK8s already uses a ``BootstrapConfiguration`` object, stored in
+``/etc/metalk8s/bootstrap.yaml``, to define how the cluster should be
+configured from the bootstrap node, and what versions of MetalK8s are available
+to the cluster.
+
+In the same vein, we want to use a ``SolutionsConfiguration`` object, stored in
+``/etc/metalk8s/solutions.yaml``, to declare which Solutions are available to
+the cluster, from the bootstrap node.
+
+.. todo:: Add specification in a future Reference guide
+
+Here is how it could look::
+
+    apiVersion: metalk8s.scality.com/v1alpha1
+    kind: SolutionsConfiguration
+    solutions:
+      - /solutions/storage_1.0.0.iso
+      - /solutions/storage_latest.iso
+      - /other_solutions/computing.iso
+
+There would be no explicit information about what the archives contain.
+Instead, we want the archive itself to contain such information (more
+details in :doc:`./archive`), and to discover it at import time.
+
+Responsibilities of each party
+------------------------------
+
+This section intends to define the boundaries between MetalK8s and the
+Solutions to integrate with, in terms of "who is doing what?".
+
+.. note:: This is still a work in progress.
+
+MetalK8s
+^^^^^^^^
+
+**MUST**:
+
+- Handle reading and mounting of the Solution ISO archive
+- Provide tooling to deploy/upgrade a Solution's CRDs and UI
+
+**MAY**:
+
+- Provide tooling to deploy/upgrade a Solution's Operator
+- Provide tooling to verify signatures in a Solution ISO
+- Expose management of Solutions in its own UI
+
+Solution
+^^^^^^^^
+
+**MUST**:
+
+- Comply with the standard archive structure defined by MetalK8s
+- Expose management of its Operator instances in its UI
+- Handle monitoring of its own services (both Operator and application, except
+  the UI)
+
+**MAY**:
+
+- Use MetalK8s monitoring services (Prometheus and Grafana)
+
+.. note::
+
+   Solutions can leverage the ``prometheus-operator`` CRs for setting up the
+   monitoring of their components.
+
+.. todo:: Define how Solutions can deploy Grafana dashboards.
+
+Interaction diagrams
+--------------------
+
+We include a detailed interaction sequence diagram for describing how MetalK8s
+will handle user input when deploying / upgrading Solutions.
```

Comment:
Would be worth mentioning rollback (even if it's exactly the same as upgrading) for operations-minded readers.

Reply:
We will add more in-depth guides for operational matters w.r.t Solutions, this one is for developers to understand how MetalK8s manages Solutions (which needs improvements as well, granted).

Most importantly, this diagram shows how deployment/upgrade is actually left to the user (manually or through the Solution UI). Rollback would fall into this category as well, IMHO.

Agreed though, we will need to show how a Solution gets **removed** from MetalK8s (which we won't do in the early stages, would be implemented in the generic `import_all_solutions` runner).

Reply:
I added a warning for removal, and mentioned more details about how an Operator is deployed manually by the user.